### PR TITLE
fix: add Ory Network login

### PR DESCRIPTION
**Pull request by vinckr**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -262,30 +215,49 @@ func pollSlack() {
...
+
+	if r.Method != http.MethodPost {
+		http.Error(w, "Method not allowed", http.StatusMethodNotAllowed)
+		return
+	}
+	// Read the request body
+	body, err := ioutil.ReadAll(r.Body)
+	if err != nil {
+		http.Error(w, "Error reading request body",
+			http.StatusInternalServerError)
```
Comment (aeneasr) [inline main.go:230]: missin return

File: static/whoami.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  } catch (error) {
+    console.error(
+      "Error during session retrieval and form submission: ",
+      error
+    );
+  }
+}
+async function getSession() {
+  const url = "https://project.console.ory.sh/sessions/whoami";
+  console.log("getting session data");
```
Comment (aeneasr) [inline static/whoami.js:14]: remove console.logs

**Comment by aeneasr** [comment]
Can you please show a screen recording of this in action?

**Comment by vinckr** [comment]
https://github.com/ory/slackinviter/assets/55394283/6925ec4b-5598-4496-b6ad-e21df1ab1c05

obviously not using the console.ory cookie and the POST to Slack is not set up correctly.

**Comment by aeneasr** [comment]
The redirect shouldn't point to /ui/welcome though should it? We should end up back at slack.ory.sh?

**Comment by vinckr** [comment]
The link on the "redirect" page is pointing to 

`https://console.ory.sh/login?return_to=https://slack.ory.sh/`

this should return you to https://slack.ory.sh/ after logging in if the subdomain is whitelisted, right? 

This is not working in the video also, but it works on my test domain

**Comment by vinckr** [comment]
accidental close sorry

**Comment by aeneasr** [comment]
> The link on the "redirect" page is pointing to
> 
> `https://console.ory.sh/login?return_to=https://slack.ory.sh/`
> 
> this should return you to https://slack.ory.sh/ after logging in if the subdomain is whitelisted, right?
> 
> This is not working in the video also, but it works on my test domain



I don't know, have you tried if it works? If not, you can find it out yourself by clicking the link ;)

**Comment by vinckr** [comment]
> I don't know, have you tried if it works? If not, you can find it out yourself by clicking the link ;)

Right it now it doesnt redirect me after login when I open https://console.ory.sh/login?return_to=https://slack.ory.sh/ . I guess the url is not allowed?

In my projects I added it here:

```
  "selfservice": {
    "allowed_return_urls": [
     .....
      "https://slack.ory.sh/"
    ],
```

can you check if that is the case for console or point me to where I can check it?

**Comment by aeneasr** [comment]
@LBBO @zepatrik please check that this includes the URLs you need.

Please don't add wildcards to redirect URLs. It allows for open redirect vulnerabilities and since we have many subdomains, we're vulnerable to subdomain takeover. Thanks!

**Comment by aeneasr** [comment]
https://github.com/ory-corp/cloud/pull/4957

**Comment by vinckr** [comment]
it works now, it needed returnTo instead of return_to :sweat:

https://console.ory.sh/login?returnTo=https://slack.ory.sh/
https://console.ory.sh/registration?returnTo=https://slack.ory.sh/

**Comment by aeneasr** [comment]
Ok, very nice! Let's go 🚢

**Pull request merged**